packages/gatsby/src/commands/build-html.js

```diff
@@ -1,69 +1,76 @@
 /* @flow */
 const webpack = require(`webpack`)
 const fs = require(`fs`)
-const debug = require(`debug`)(`gatsby:html`)
 
 const webpackConfig = require(`../utils/webpack.config`)
-const { store } = require(`../redux`)
 const { createErrorFromString } = require(`gatsby-cli/lib/reporter/errors`)
 const renderHTMLQueue = require(`../utils/html-renderer-queue`)
 const telemetry = require(`gatsby-telemetry`)
 
-module.exports = async (program: any, activity: any) => {
+const runWebpack = compilerConfig =>
+  new Promise((resolve, reject) => {
+    webpack(compilerConfig).run((e, stats) => {
+      if (e) {
+        reject(e)
+      } else {
+        resolve(stats)
+      }
+    })
+  })
+
+const doBuildRenderer = async (program, webpackConfig) => {
   const { directory } = program
+  const stats = await runWebpack(webpackConfig)
+  // render-page.js is hard coded in webpack.config
+  const outputFile = `${directory}/public/render-page.js`
+  if (stats.hasErrors()) {
+    let webpackErrors = stats.toJson().errors.filter(Boolean)
+    const error = webpackErrors.length
+      ? createErrorFromString(webpackErrors[0], `${outputFile}.map`)
+      : new Error(
+          `There was an issue while building the site: ` +
+            `\n\n${stats.toString()}`
+        )
+    throw error
+  }
+  return outputFile
+}
+
+const buildRenderer = async (program, stage) => {
+  const { directory } = program
+  const config = await webpackConfig(program, directory, stage, null)
+  return await doBuildRenderer(program, config)
```

Comment:
no need to await here as you await in buildPages
```suggestion
  return doBuildRenderer(program, config)
```

Reply:
In general, I like to explicitly `await` on async functions, since it's an extra hint to the reader that `doBuildRenderer` is an async function. As you say, it's very much redundant, but the positive is a little more clarity in the code. I don't feel strongly either way though, so happy to change it.

Reply:
I think both are fine. I write my code like @wardpeet does here--but I also see your point in clarifying intent, which I kinda like!

I wouldn't necessarily feel like you have to take this suggestion--but thanks for being receptive to it!

Reply:
just a suggestion, so no need to change :) thanks for responding